packages/cardano-services/src/Asset/CardanoTokenRegistry.ts

```diff
@@ -0,0 +1,56 @@
+import { Asset } from '@cardano-sdk/core';
+import { Logger } from 'ts-log';
+import { TokenMetadataService } from './types';
+import axios from 'axios';
+
+/**
+ * Dependencies that are need to create CardanoTokenRegistry
+ */
+export interface CardanoTokenRegistryDependencies {
+  logger: Logger;
+}
+
+/**
+ * TokenMetadataService implementation using Cardano Token Registry public API
+ */
+export class CardanoTokenRegistry implements TokenMetadataService {
+  #cache: Record<string, Asset.TokenMetadata> = {};
+  #logger: Logger;
+
+  constructor({ logger }: CardanoTokenRegistryDependencies) {
+    this.#logger = logger;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  #checkNumberValue(data: any, attribute: string) {
+    if (attribute in data && typeof data[attribute].value === 'number') return data[attribute].value as number;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  #checkStringValue(data: any, attribute: string) {
+    if (attribute in data && typeof data[attribute].value === 'string') return data[attribute].value as string;
+  }
+
+  async getTokenMetadata(subject: string): Promise<Asset.TokenMetadata | null> {
+    if (this.#cache[subject]) return this.#cache[subject];
+
+    this.#logger.debug(`Requesting for metatada for subject "${subject}"`);
+    const metadata: Asset.TokenMetadata = { version: '1.0' };
```

Comment:
Can probably just omit this hard-coded value given it's the default: https://github.com/input-output-hk/cardano-js-sdk/blob/1a873ec182c901813feae8244c86c0346bb70022/packages/core/src/Asset/types/TokenMetadata.ts#L58